CMSgov/mdct-mcr
# clean and organize frontend

## Description

The frontend was messy and unorganized, and now it is clean and ready for new dev.

Note: This ended up being a huge PR with lots of files touched, which I know violates the first rule of PRs. Forgive me. Lucky for us, there's very little functionality in MCR right now, so there's also very little to break. The changes are mostly cleanup, formatting, organizing, and renaming, so don't lose too much sleep over this one.

### How to test

No functional changes -- just check that current behavior works as expected.

1. Check that your local envs match the following: 
     - root env:
       ```
       SKIP_PREFLIGHT_CHECK=true
       LOCAL_LOGIN=true
       MEASURE_TABLE_NAME=local-measures
       MEASURE_TABLE_ARN=local_nonsense_if_unset_we_search_CF_for
       measureTableName=local-measures
       DYNAMODB_URL=http://localhost:8000/
       API_URL=http://localhost:3030/local
       S3_LOCAL_ENDPOINT=http://localhost:4569/
       S3_ATTACHMENTS_BUCKET_NAME=local-uploads
       URL=http://localhost/3000
       ```
     - services/ui-src env:
       ```
       LOCAL_LOGIN=true
       API_REGION=us-east-1
       API_URL=http://localhost:3030/
       COGNITO_REGION=us-east-1
       COGNITO_IDENTITY_POOL_ID=us-east-1:61c849d8-f95c-45af-8131-15b75c525034
       COGNITO_USER_POOL_ID=us-east-1_7U4ihGYZP
       COGNITO_USER_POOL_CLIENT_ID=46fh6sftchq768ab0v65cpq0cf
       COGNITO_USER_POOL_CLIENT_DOMAIN=master-login-46fh6sftchq768ab0v65cpq0cf.auth.us-east-1.amazoncognito.com
       COGNITO_REDIRECT_SIGNIN=http://localhost:3000/
       COGNITO_REDIRECT_SIGNOUT=http://localhost:3000/
       S3_ATTACHMENTS_BUCKET_REGION=us-east-1
       S3_ATTACHMENTS_BUCKET_NAME=uploads-master-attachments-024259748323
       S3_LOCAL_ENDPOINT=http://localhost:4569/
       ```
3. Run `./dev local`
4. Sign in with the following credentials: 
     - un: stateuser2@test.com
     - pw: p@55W0rd!
5. Click `Create Measure`. Expect: Some fake data to appear.
6. Click `Delete Measure`. Expect: That fake data to disappear.

Can also run the unit tests if you feel like it: 
1. `cd services/ui-src`
3. `yarn test`

### Dependencies

No changed deps.

## Code author checklist
- [x] I have added [thorough tests](https://qmacbis.atlassian.net/wiki/spaces/CM/pages/2914025525/Test+Suite+and+Testing+Research)

## Reviewer checklist (two different people)
